mpags-cipher.cpp

```diff
@@ -0,0 +1,40 @@
+#include <iostream>
+#include <string>
+#include <vector>
+
+int main(int argc, char* argv[])
+{
+    const std::vector<std::string> cmdLineArgs{argv,argv+argc};
+
+    for(std::size_t i{0};i<cmdLineArgs.size();i++)
+    { 
+
+        if(cmdLineArgs[i] == "-h" or cmdLineArgs[i] == "--help")
+        {
+            std::cout << "Help Text" << std::endl;
+            continue;
+        }
+        
+        if(cmdLineArgs[i] == "--version")
+        {
+            std::cout << "version 1.0.0" << std::endl;
+            continue;
+        }
+        
+        if(cmdLineArgs[i] == "-i")
+        {
+            std::cout << "Input File:   " << cmdLineArgs[i+1] << std::endl;
+            i++;
+            continue;
```

Comment:
You've handled the option well with `i++` but technically you should have stored the value specified in a `string` for later.

Also, there is no check for if the `-i` option is the last one, e.g.
```
./mpags-cipher -i
```
This should print an error if so.